app/Models/Token.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace Adshares\Adserver\Models;
+
+use Adshares\Adserver\Events\GenerateUUID;
+use Adshares\Adserver\Models\Traits\AutomateMutators;
+use Adshares\Adserver\Models\Traits\BinHex;
+use Adshares\Adserver\Models\Traits\Serialize;
+use Illuminate\Database\Eloquent\Model;
+
+class Token extends Model
+{
+    use AutomateMutators;
+    use BinHex;
+    use Serialize;
+
+    /**
+     * The event map for the model.
+     *
+     * @var array
+     */
+    protected $dispatchesEvents = [
+        'creating' => GenerateUUID::class,
+    ];
+
+    /**
+     * The attributes that are mass assignable.
+     *
+     * @var array
+     */
+    protected $fillable = [
+        'multi_usage', 'payload', 'tag', 'user_id', 'valid_until',
+    ];
+
+    /**
+     * The attributes that use some Models\Traits with mutator settings automation.
+     *
+     * @var array
+     */
+    protected $traitAutomate = [
+        'uuid' => 'BinHex',
+        'payload' => 'Serialize',
+    ];
+
+    /**
+     * Checks if can generate another token for user based on time limit and tag.
+     *
+     * @param int    $user_id
+     * @param string $tag
+     * @param int    $older_then_seconds
+     *
+     * @return bool
+     */
+    public static function canGenerate(int $user_id, $tag, int $older_then_seconds)
+    {
+        if (self::where('user_id', $user_id)->where('tag', $tag)->where('created_at', '>', date('Y-m-d H:i:s', time() - $older_then_seconds))->count()) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Y-m-d H:i:s" 5 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1192)